src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory_Methods.vb

```diff
@@ -23,7 +23,7 @@ Namespace Microsoft.CodeAnalysis.Semantics
                         ' It is not permissible to access the Left property of a BoundUserDefinedBinaryOperator unconditionally,
                         ' because that property can throw an exception if the operator expression is semantically invalid.
                         ' Fetching the left operand through IBinaryOperatorExpression is safe.
-                        If rightOperatorBinary.LeftOperand Is value.LeftOnTheRightOpt Then
+                        If rightOperatorBinary.LeftOperand Is Create(value.LeftOnTheRightOpt) Then
```

Comment:
It feels like all these checks should be performed on BoundNodes rather than on IOperation nodes. See how it is done in the previous case. I do not understand how did we end-up with code like this. I was under impression that the code was simply moved, but it turned out this is not the case, a projection to IOperation was added in some cases. Please identify all places like this in for both languages and remove added projections. 

---
In reply to: [116078819](https://github.com/dotnet/roslyn/pull/19406#discussion_r116078819) [](ancestors = 116078819)